src/Symfony/Component/Validator/Mapping/MemberMetadata.php

```diff
@@ -29,42 +29,30 @@
  */
 abstract class MemberMetadata extends GenericMetadata implements PropertyMetadataInterface
 {
-    /**
-     * @internal This property is public in order to reduce the size of the
```

Comment:
We lose the `@internal` flag with this change.

Reply:
Loosing the `@internal` tag is a no-go as it would mean those public properties become covered by the BC promise of Symfony.

Reply:
Oh you're right, I added the comment in the constructor but forget the `@internal` flag. Do you think this will work ?

```php
/**
 * @param string $class    The name of the class this member is defined on
 * @internal               This property is public in order to reduce the size of the
 *                         class' serialized representation. Do not access it. Use
 *                         {@link getClassName()} instead.
 * ...
```

Reply:
Should I try to put the comment in the constructor as suggested (don't know if it works) or remove this class from this PR?

Reply:
I'd remove the change from the PR.

Reply:
@AurelienPillevesse no. This won't work as the `@internal` would apply to the constructor, and the constructor is not internal.

This is a case where promoted properties should not be used.

Reply:
Fixed, sorry didn't know about it